pkg/controller/job/job_controller.go

```diff
@@ -143,6 +144,13 @@ type syncJobCtx struct {
 	ready                           int32
 }
 
+type orphanPodKey struct {
+	// Either "key" or "selector"
+	kind string
```

Comment:
I don't feel strongly about it, but I would prefer to have enum consts for this. I think then we can get support for static code analysis for the `switch` statement in code.